### PR TITLE
Fix Composer not focusing on edit/reply

### LINKED ISSUE #17599: The composer is not activated for input after trying to edit or reply to a message
### Steps to Reproduce:
1. Open a chat with existing messages.
2. Attempt to edit or reply to a message by tapping on it.
3. Observe the behavior of the composer for input.

### Actual Result:
After trying to edit or reply to a message, the composer for input is not activated, and users cannot immediately start typing a new message.

https://github.com/status-im/status-mobile/assets/52490791/c314c535-f5c7-4f3d-99da-893cc6fe57be

### Expected Result:
The composer should be automatically activated for input after attempting to edit or reply to a message, allowing users to start typing a new message without extra steps.

### Additional Information
- Status version: nightly [10 Oct 2023]
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #17599

### Summary

There's a common issue with RN that affects how we trigger the focusing of text input elements and it prevented composer from switching to the focused state when the user wanted to reply/edit.
We needed to schedule the focusing action for it to work, This PR adds that fix.
 
### Before and after video comparison

| Before | After
| --- | --- |
| CleanShot 2023-10-15 at 11 53 16 | CleanShot 2023-10-15 at 11 50 56  |

status: ready

### CODE REVIEW

**Review by OmarBasem — APPROVED** [review]

**Review by Parveshdhull — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
Hi @ibrkhalil, thank you for PR. The issue is still reproducible in 20% of cases for Android devices:
### Steps to Reproduce:
1. Open a chat with existing messages.
2. Attempt to edit or reply to a message by tapping on it.
3. Observe the behavior of the composer for input.

### Actual result:
The composer is not activated for input after trying to edit or reply to a message

https://github.com/status-im/status-mobile/assets/52490791/5c558242-e0a6-41a8-9e9f-3dec4c09d85a

### Expected result:
The composer should be automatically activated for input after attempting to edit or reply to a message, allowing users to start typing a new message without extra steps.

### Devices:
- Android emulator: Pixel 6 Pro API 31
- Real device: Pixel 7a, Android 13

**Comment by ibrkhalil** [comment]
> Hi @ibrkhalil, thank you for PR. The issue is still reproducible in 20% of cases for Android devices:
> 
> ### Steps to Reproduce:
> 1. Open a chat with existing messages.
> 2. Attempt to edit or reply to a message by tapping on it.
> 3. Observe the behavior of the composer for input.
> 
> ### Actual result:
> The composer is not activated for input after trying to edit or reply to a message
> 
>  not_opened_textbox.mp4 
> ### Expected result:
> The composer should be automatically activated for input after attempting to edit or reply to a message, allowing users to start typing a new message without extra steps.
> 
> ### Devices:
> * Android emulator: Pixel 6 Pro API 31
> * Real device: Pixel 7a, Android 13

Got it, I think we need to increase the delay a little bit more so that it can work for even older devices.
I am going to adjust the value now and please tell me if it's enough

**Comment by ibrkhalil** [comment]
> Hi @ibrkhalil, thank you for PR. The issue is still reproducible in 20% of cases for Android devices:
> ### Steps to Reproduce:
> 1. Open a chat with existing messages.
> 2. Attempt to edit or reply to a message by tapping on it.
> 3. Observe the behavior of the composer for input.
> 
> ### Actual result:
> The composer is not activated for input after trying to edit or reply to a message
> 
> https://github.com/status-im/status-mobile/assets/52490791/5c558242-e0a6-41a8-9e9f-3dec4c09d85a
> 
> ### Expected result:
> The composer should be automatically activated for input after attempting to edit or reply to a message, allowing users to start typing a new message without extra steps.
> 
> ### Devices:
> - Android emulator: Pixel 6 Pro API 31
> - Real device: Pixel 7a, Android 13

Done

**Comment by VolodLytvynenko** [comment]
Hi @ibrkhalil, thank you for PR. The issue is still reproducible in 15% of cases for Android devices during messaeg editing and every time during reply:

Editing:

https://github.com/status-im/status-mobile/assets/52490791/f0bbd5ae-fc10-4794-bce0-06bb0a30080f

Reply:

https://github.com/status-im/status-mobile/assets/52490791/1fc22953-af66-424c-8c60-b2558cb5d863


### Devices:
- Android emulator: Pixel 6 Pro API 31
- Real device: Pixel 7a, Android 13

**Comment by VolodLytvynenko** [comment]
> Hi @ibrkhalil, thank you for PR. The issue is still reproducible in 15% of cases for Android devices during messaeg editing and every time during reply:
> 
> Editing:
> 
>  editing.mp4 
> Reply:
> 
>  reply.mp4 
> ### Devices:
> * Android emulator: Pixel 6 Pro API 31
> * Real device: Pixel 7a, Android 13

@ibrkhalil it seems like this issue is reproducible for specific devices. For instance, I couldn't replicate it on the Android emulator: Pixel XL API 33, but I could easily reproduce it on the:  Android emulator: Pixel 6 Pro API 31 and on a real device, the Pixel 7a running Android 13.

**Comment by ibrkhalil** [comment]
@VolodLytvynenko Should be working now.

File: src/status_im2/contexts/chat/composer/effects.cljs [hunk 1]
```diff
@@ -112,7 +112,10 @@
      (let [edit-text        (get-in edit [:content :text])
            text-value-count (count @text-value)]
        (when (and edit @input-ref)
-         (.focus ^js @input-ref)
+         ;; A small setTimeout is necessary to ensure the statement is enqueued and will get executed
+         ;; ASAP.
+         ;; https://github.com/software-mansion/react-native-screens/issues/472
+         (js/setTimeout #(.focus ^js @input-ref) 250)
```
Comment (briansztamfater) [inline src/status_im2/contexts/chat/composer/effects.cljs:118]: We should make sure timers are cleared when the component unmounts

Comment (ibrkhalil) [reply src/status_im2/contexts/chat/composer/effects.cljs:118]: Shouldn't timeouts get cleared automatically when they finish?
I thought the usage of clearTimeout was only to abort the execution of the callback we pass to setTimeout.

**Comment by ibrkhalil** [comment]
@status-im/mobile-qa 
Ready for review.

**Comment by VolodLytvynenko** [comment]
@ibrkhalil issues are fixed. PR is tested and ready to be merged. Thanx!

**Pull request merged**